src/cpp/fastdds/subscriber/history/DataReaderInstance.hpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DataReaderInstance.hpp
+ */
+
+#ifndef _FASTDDS_SUBSCRIBER_HISTORY_DATAREADERINSTANCE_HPP_
+#define _FASTDDS_SUBSCRIBER_HISTORY_DATAREADERINSTANCE_HPP_
+
+#include <chrono>
+#include <cstdint>
+#include <map>
+
+#include <fastdds/dds/subscriber/InstanceState.hpp>
+#include <fastdds/dds/subscriber/ViewState.hpp>
+
+#include <fastrtps/utils/collections/ResourceLimitedVector.hpp>
+
+#include "DataReaderCacheChange.hpp"
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace detail {
+
+/// Book-keeping information for an instance
+struct DataReaderInstance
+{
+    using ChangeCollection = eprosima::fastrtps::ResourceLimitedVector<DataReaderCacheChange, std::true_type>;
+    using WriterCollection = std::map<fastrtps::rtps::GUID_t, uint32_t>;
```

Comment:
The use of a std::map breaks the zero dynamic allocation policy.

Reply:
Changed into a ResourceLimitedVector of pairs